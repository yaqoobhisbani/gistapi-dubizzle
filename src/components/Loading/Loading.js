import React, { useEffect, useState } from "react";
import styled from "styled-components";

function Loading() {
  const [text, setText] = useState("Loading");

  useEffect(() => {
    const interval = setInterval(() => {
      setText((prev) => {
        if (prev.includes("...")) return "Loading";
        else return prev + ".";
      });
    }, 500);

    return () => {
      clearInterval(interval);
    };
  }, []);

  return (
    <Wrapper>
      <LoadingText>{text}</LoadingText>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const LoadingText = styled.h3`
  font-size: 16px;
  color: ${(props) => props.theme.primary};
  font-weight: normal;
`;

export default Loading;
