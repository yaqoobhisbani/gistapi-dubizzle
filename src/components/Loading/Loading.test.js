import React from "react";
import { render } from "@testing-library/react";
import Loading from "./Loading";

describe("Loading", () => {
  test("renders loading text", () => {
    const { getByText } = render(<Loading />);
    const textElement = getByText("Loading");
    expect(textElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
