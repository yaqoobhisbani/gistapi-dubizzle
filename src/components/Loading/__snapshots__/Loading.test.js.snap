// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB bphukn"
  >
    <h3
      class="sc-gueYoa genrlf"
    >
      Loading
    </h3>
  </div>
</div>
`;
